docs/readmes/proposals/p010_vendor_neutral_dp.md

```diff
@@ -416,7 +456,250 @@ scope of work.
+  having one of them in place, it should be straightforward to add
+  another one (it can be achieved by having a ProtocolController abstract class
+  to provide structure for particular PC implementations).
+  
+  To scale the system better we are decoupling PC (responsible for protocol)
+  from Radio controller (responsible for the business domain).
+  Additional benefit is maintaining all of the protocol customizations
+  and tricks at the PC level and making the PC layer scalable
+  according to the number of eNBs of each type.
+
+* **Radio Controller (RC)**
+  
+  RC is responsible for the CBSD to SAS communication
+  abstracted from vendor specific protocols.
+  
+  RC on one hand is responsible for placing the requests coming from PCs in the queue
+  and on the other for segregating the responses coming from Configuration Controller
+  and sending them back to the appropriate PC.
+
+* **Configuration Controller (CC)**
+
+  Configuration Controller is responsible for polling requests periodically from the queue,
+  aggregating them into one request in format understandable by SAS,
+  posting them over to SAS
+  and once a response is received, passing it on to RC.
+
+* **Message Queue**
+
+  RabbitMQ instance for storage of requests received by the RC
+  allowing them to be polled by a consumer.
+  Requests for which acknowledgement is received upon polling
+  will be deleted from the queue,
+  while the ones without acknowledgement will be requeued.
+
+* **Database (optional, to be implemented at stage 3)**
+
+  Relational DB storing the state of the DP.
+
+  DB allows to keep track of grants and CBSDs configuration limiting redundant CBSD - SAS connections.
+
+## Component Interactions
+
+* **Protocol Controller**
+
+  PC will pass the requests it receives from CBSDs to the RC over HTTPS.
+  
+  PC will pass the responses that are coming from RC back to the respective CBSDs
+  over the protocol that they are able to understand and the PC is implementing.
+
+* **Radio Controller**
+
+  RC receives requests from the PC over HTTPs and puts them in
+  the queue over AMQP.
+
+  When the response for the request comes back from CC,
+  RC will forward it over HTTPS to the appropriate PC.
+
+* **Configuration Controller**
+
+  CC communicates with the queue over AMQP for request polling.
+  
+  CC communicates with SAS (external) over HTTPS for posting bulked requests and with the RC over HTTPS for forwarding the SAS response.
+
+* **Message Queue**
+
+  Treating RC as the publisher and CC as the consumer.
+
+* **Database**
+
+  Exchanging information about CBSDs and grants with the CC.
+
+## Solution scaling
+
+The Domain Proxy will be split into separate services in separate containers.
+
+Every service will work in HA mode to provide service even in failover scenarios.
+
+Components such as database and queue will use cluster solutions
+delivered by database and queue providers.
+
+All other components will be stateless Python based services
```

Comment:
minor point: All services on orc8r are golang based. Might be easier to integrate in homogenous environment.

Reply:
Also you can get services like configurator, state management etc for free

Reply:
With well defined APIs between microservices the implementation language becomes a secondary thing.
We have chosen python for a few reasons:
* Community uses Python already
* Test suite which we use for certifying the solution is entirely python
* high python expertise level of the dev team